internal/fs/vss_windows.go

```diff
@@ -0,0 +1,977 @@
+	result, _, _ := syscall.Syscall(vss.getVTable().abortBackup, 1,
+		uintptr(unsafe.Pointer(vss)), 0, 0)
+
+	return newVssErrorIfResultNotOK("AbortBackup() failed", HRESULT(result))
+}
+
+// InitializeForBackup calls the equivalent VSS api.
+func (vss *IVssBackupComponents) InitializeForBackup() error {
+	result, _, _ := syscall.Syscall(vss.getVTable().initializeForBackup, 2,
+		uintptr(unsafe.Pointer(vss)), 0, 0)
+
+	return newVssErrorIfResultNotOK("InitializeForBackup() failed", HRESULT(result))
+}
+
+// SetContext calls the equivalent VSS api.
+func (vss *IVssBackupComponents) SetContext(context VssContext) error {
+	result, _, _ := syscall.Syscall(vss.getVTable().setContext, 2,
+		uintptr(unsafe.Pointer(vss)), uintptr(context), 0)
+
+	return newVssErrorIfResultNotOK("SetContext() failed", HRESULT(result))
+}
+
+// GatherWriterMetadata calls the equivalent VSS api.
+func (vss *IVssBackupComponents) GatherWriterMetadata() (*IVSSAsync, error) {
+	var oleIUnknown *ole.IUnknown
+	result, _, _ := syscall.Syscall(vss.getVTable().gatherWriterMetadata, 2,
+		uintptr(unsafe.Pointer(vss)), uintptr(unsafe.Pointer(&oleIUnknown)), 0)
+
+	err := newVssErrorIfResultNotOK("GatherWriterMetadata() failed", HRESULT(result))
+	return vss.convertToVSSAsync(oleIUnknown, err)
+}
+
+// convertToVSSAsync looks up IVSSAsync interface if given result
+// is a success.
+func (vss *IVssBackupComponents) convertToVSSAsync(
+	oleIUnknown *ole.IUnknown, err error) (*IVSSAsync, error) {
+	if err != nil {
+		return nil, err
+	}
+
+	comInterface, err := queryInterface(oleIUnknown, UIID_IVSS_ASYNC)
+	if err != nil {
+		return nil, err
+	}
+
+	iVssAsync := (*IVSSAsync)(unsafe.Pointer(comInterface))
+	return iVssAsync, nil
+}
+
+// IsVolumeSupported calls the equivalent VSS api.
+func (vss *IVssBackupComponents) IsVolumeSupported(volumeName string) (bool, error) {
+	volumeNamePointer, err := syscall.UTF16PtrFromString(volumeName)
+	if err != nil {
+		panic(err)
+	}
+
+	var isSupportedRaw uint32
+	var result uintptr
+
+	if runtime.GOARCH == "386" {
+		id := (*[4]uintptr)(unsafe.Pointer(ole.IID_NULL))
+
+		result, _, _ = syscall.Syscall9(vss.getVTable().isVolumeSupported, 7,
+			uintptr(unsafe.Pointer(vss)), id[0], id[1], id[2], id[3],
+			uintptr(unsafe.Pointer(volumeNamePointer)), uintptr(unsafe.Pointer(&isSupportedRaw)), 0,
+			0)
+	} else {
+		result, _, _ = syscall.Syscall6(vss.getVTable().isVolumeSupported, 4,
+			uintptr(unsafe.Pointer(vss)), uintptr(unsafe.Pointer(ole.IID_NULL)),
+			uintptr(unsafe.Pointer(volumeNamePointer)), uintptr(unsafe.Pointer(&isSupportedRaw)), 0,
+			0)
+	}
+
+	var isSupported bool
+	if isSupportedRaw == 0 {
+		isSupported = false
+	} else {
+		isSupported = true
+	}
+
+	return isSupported, newVssErrorIfResultNotOK("IsVolumeSupported() failed", HRESULT(result))
+}
+
+// StartSnapshotSet calls the equivalent VSS api.
+func (vss *IVssBackupComponents) StartSnapshotSet() (ole.GUID, error) {
+	var snapshotSetId ole.GUID
```

Comment:
golint: var snapshotSetId should be snapshotSetID

Reply:
Fixed in https://github.com/restic/restic/pull/2274/commits/f70a1dfd172d54843e27deae292f3557d11ade84